File: app/src/main/java/jp/juggler/fadownloader/PageOther.java
package jp.juggler.fadownloader;

import android.app.Activity;
import android.content.Intent;
import android.location.Location;
import android.net.Uri;
import android.provider.Settings;
import android.view.View;

import java.util.Locale;

public class PageOther extends PagerAdapterBase.PageViewHolder implements View.OnClickListener{

	View btnRemoveAd;

	public PageOther( Activity activity, View ignored ){
		super( activity, ignored );
	}

	@Override protected void onPageCreate( int page_idx, View root ) throws Throwable{
		btnRemoveAd = root.findViewById( R.id.btnRemoveAd );

		root.findViewById( R.id.btnLogClear ).setOnClickListener( this );
		root.findViewById( R.id.btnOSSLicence ).setOnClickListener( this );
		root.findViewById( R.id.btnWifiSetting ).setOnClickListener( this );
		root.findViewById( R.id.btnOpenMap ).setOnClickListener( this );
		root.findViewById( R.id.btnRecordClear ).setOnClickListener( this );
		root.findViewById( R.id.btnRemoveAd ).setOnClickListener( this );

		updatePurchaseButton();
	}

	@Override protected void onPageDestroy( int page_idx, View root ) throws Throwable{
	}

	@Override public void onClick( View view ){
		switch( view.getId() ){

		case R.id.btnLogClear:
			activity.getContentResolver().delete( LogData.meta.content_uri, null, null );
			break;

		case R.id.btnRecordClear:
			activity.getContentResolver().delete( DownloadRecord.meta.content_uri, null, null );
			break;


		case R.id.btnOSSLicence:
			( (ActMain) activity ).openHelp( activity.getString( R.string.help_oss_license_long ) );
			break;

		case R.id.btnRemoveAd:
			( (ActMain) activity ).startRemoveAdPurchase();
			break;

		case R.id.btnWifiSetting:
			try{
				activity.startActivity( new Intent( Settings.ACTION_WIFI_SETTINGS ) );
			}catch( Throwable ex ){
				ex.printStackTrace();
			}
			break;

		case R.id.btnOpenMap:
			try{
				Location location = DownloadService.getLocation();
				if( location != null){
					Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(String.format( Locale.JAPAN,"geo:%f,%f",location.getLatitude(),location.getLongitude())));
					activity.startActivity( intent );
				}else{
					Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse("geo:"));
					activity.startActivity( intent );
				}
			}catch( Throwable ex ){
				ex.printStackTrace();
			}
			break;

		}
	}

	public void updatePurchaseButton(){
		ActMain act = (ActMain) activity;
		//noinspection ConstantConditions
		btnRemoveAd.setVisibility( act.bSetupCompleted && ! act.bRemoveAdPurchased ? View.VISIBLE : View.GONE );
	}
}

File: app/src/main/java/jp/juggler/fadownloader/PageOther.kt
package jp.juggler.fadownloader

import android.app.Activity
import android.content.Intent
import android.net.Uri
import android.provider.Settings
import android.view.View
import jp.juggler.fadownloader.table.DownloadRecord
import jp.juggler.fadownloader.util.LogTag
import jp.juggler.fadownloader.util.PagerAdapterBase
import java.util.*

class PageOther(activity : Activity, ignored : View) :
	PagerAdapterBase.PageViewHolder(activity, ignored), View.OnClickListener {
	
	companion object {
		private val log = LogTag("PageOther")
	}
	private lateinit var btnRemoveAd : View
	
	@Throws(Throwable::class)
	override fun onPageCreate(page_idx : Int, root : View) {
		btnRemoveAd = root.findViewById(R.id.btnRemoveAd)
		
		root.findViewById<View>(R.id.btnOSSLicence).setOnClickListener(this)
		root.findViewById<View>(R.id.btnWifiSetting).setOnClickListener(this)
		root.findViewById<View>(R.id.btnOpenMap).setOnClickListener(this)
		root.findViewById<View>(R.id.btnRecordClear).setOnClickListener(this)
		root.findViewById<View>(R.id.btnRemoveAd).setOnClickListener(this)
		
		updatePurchaseButton()
	}
	
	@Throws(Throwable::class)
	override fun onPageDestroy(page_idx : Int, root : View) {
	}
	
	override fun onClick(view : View) {
		when(view.id) {
			
			R.id.btnRecordClear -> activity.contentResolver.delete(
				DownloadRecord.meta.content_uri,
				null,
				null
			)
			
			R.id.btnOSSLicence -> (activity as ActMain).openHelp(activity.getString(R.string.oss_license_long_help))
			
			R.id.btnRemoveAd -> (activity as ActMain).startRemoveAdPurchase()
			
			R.id.btnWifiSetting -> try {
				activity.startActivity(Intent(Settings.ACTION_WIFI_SETTINGS))
			} catch(ex : Throwable) {
				log.trace(ex,"btnWifiSetting failed")
			}
			
			R.id.btnOpenMap -> try {
				val location = DownloadService.location
				if(location != null) {
					val intent = Intent(
						Intent.ACTION_VIEW,
						Uri.parse(
							String.format(
								Locale.JAPAN,
								"geo:%f,%f",
								location.latitude,
								location.longitude
							)
						)
					)
					activity.startActivity(intent)
				} else {
					val intent = Intent(Intent.ACTION_VIEW, Uri.parse("geo:"))
					activity.startActivity(intent)
				}
			} catch(ex : Throwable) {
				log.trace(ex,"btnOpenMap failed")
			}
			
		}
	}
	
	fun updatePurchaseButton() {
		val act = activity as ActMain
		
		btnRemoveAd.visibility = if(act.bSetupCompleted && ! act.bRemoveAdPurchased) View.VISIBLE else View.GONE
	}
}
